crates/daphne/src/lib.rs

```diff
@@ -814,7 +819,25 @@ impl deepsize::DeepSizeOf for DapAggregationParam {
 }
 
 impl Encode for DapAggregationParam {
-    fn encode(&self, bytes: &mut Vec<u8>) -> Result<(), CodecError> {
+    fn encode(&self, _bytes: &mut Vec<u8>) -> Result<(), CodecError> {
+        match self {
+            Self::Empty => Ok(()),
+            #[cfg(feature = "experimental")]
+            Self::Mastic(_) => Err(CodecError::UnexpectedValue),
+        }
+    }
+
+    fn encoded_len(&self) -> Option<usize> {
+        match self {
+            Self::Empty => Some(0),
+            #[cfg(feature = "experimental")]
+            Self::Mastic(agg_param) => agg_param.encoded_len(),
+        }
+    }
+}
+
+impl EncodeDraft09 for DapAggregationParam {
+    fn encode(&self, bytes: &mut Vec<u8>) -> Result<(), CodecErrorDraft09> {
```

Comment:
This is an ideal use case for macros:

```rust
macro_rules! impl_encode_for_agg_param {
    ($encode_trait:ident, $codec_error:ident) => {
        impl $encode_trait for DapAggregationParam {
            fn encode(&self, _bytes: &mut Vec<u8>) -> Result<(), $codec_error> {
                match self {
                    Self::Empty => Ok(()),
                    #[cfg(feature = "experimental")]
                    Self::Mastic(_) => Err($codec_error::UnexpectedValue),
                }
            }

            fn encoded_len(&self) -> Option<usize> {
                match self {
                    Self::Empty => Some(0),
                    #[cfg(feature = "experimental")]
                    Self::Mastic(agg_param) => agg_param.encoded_len(),
                }
            }
        }
    };
}
impl_encode_for_agg_param!(Encode, CodecError);
impl_encode_for_agg_param!(EncodeDraft09, CodecErrorDraft09);
```

Likewise for `ParameterizedDecode` below.

Reply:
Ohhhh I see the output for the Mastic variant is different in each case. What I suggest we do is just return `Ok(Self::Mastic(Poplar1AggregationParam::decode(bytes)?))` in both instances.

Reply:
Not the experimental branch?

Reply:
You still need `#[cfg(feature = "experimental")]` because Mastic is only compiled when the "experimental" feature is enabled.